src/components/RefundBankAccount.tsx

```diff
@@ -53,7 +53,7 @@ export default function RefundBankAccount (): ReactElement {
                 label="Bank Routing number"
                 register={register}
                 required={true}
-                patternConfig={Patterns.bankRouting}
+                patternConfig={Patterns.bankRouting.get(control)}
```

Comment:
This is the main refactoring change. 

It might be better to just split numeric fields and text fields into separate components, since numeric fields now require access to the control object and text fields do not. 

On the other hand, I like having all the field configurations set up concisely in Patterns, then having one component handle the rendering logic.